### ISSUE TITLE
Error Message: error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400

### ISSUE THREAD

**Issue by Tekano985** [microsoft/vscode-azureappservice#1445]
How ive been trying to deploy my app for several times and I continue to recieve this error 

Action: appService.Deploy
Error type: Error
Error Message: error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
fatal: the remote end hung up unexpectedly
fatal: the remote end hung up unexpectedly
Everything up-to-date

Version: 0.16.3
OS: linux
OS Release: 5.3.0-40-generic
Product: Visual Studio Code
Product Version: 1.43.0
Language: en

**Comment by nturinski** [comment]
Hi @Tekano985 

Are you trying to deploy via Local Git? Could you provide repro steps?

**Comment by songpr** [comment]
I have the same issue when try to deploy using local git
steps
1) right click at Azure App Service that want to deploy
2) select: Deploy to Web App...
3) Error shown as in the post

**Comment by nturinski** [comment]
Hi @songpr

Thanks for the information.  Do you think that this could resolve your issue?

> * When pushing a large amount of data (initial push of a big repository, change with very big file(s)) may require a higher http.postBuffer setting on your git client (not the server).
> * Increase the Git buffer size to the largest individual file size of your repo:\
>   * git config --global http.postBuffer 157286400

[-Source](https://confluence.atlassian.com/bitbucketserverkb/git-push-fails-fatal-the-remote-end-hung-up-unexpectedly-779171796.html)

**Comment by songpr** [comment]
I've tried that already but it's not help

**Comment by songpr** [comment]
Hi @nturinski @Tekano985 
I found a solution. Actually , this issue cause by storage of app service plan is full.
Just upgrade the app service plan or remove some app service will help solve this issue.

The error message from Azure is just not help.

Thanks @nturinski for trying to investigate the issue.

**Comment by nturinski** [comment]
Hi @songpr

Thanks for your solution!  I'm sorry that I wasn't able to help-- I know it must have been frustrating.  But now that we know, we can remember if anyone else runs into this problem 😃

**Comment by Tekano985** [comment]
Hi, @songpr I tried it too, didnt work for me, I ended up redoing the project. and changed from using local git to using gitlab now its working perfect

**Comment by EricJizbaMSFT** [comment]
Seems like everyone on this thread has found a workaround. Closing as there's not much actionable on the VS Code side. Using a source control like bitbucket as @Tekano985 mentioned is definitely a good idea and is the best practice for production apps

I found [this doc](https://docs.microsoft.com/azure/azure-functions/functions-deployment-technologies#local-git) which actually discourages the use of local git, but that might be specific to Azure Functions and not web apps.
